src/dash-table/dash/Sanitizer.ts

```diff
@@ -62,27 +62,39 @@ const applyDefaultsToColumns = (defaultLocale: INumberLocale, defaultSort: SortA
 
 const applyDefaultToLocale = (locale: INumberLocale) => getLocale(locale);
 
+const getVisibleColumns = (
+    columns: Columns,
+    hiddenColumns: string[] | undefined
+) => R.filter(column => !hiddenColumns || hiddenColumns.indexOf(column.id) < 0, columns);
+
 export default class Sanitizer {
     sanitize(props: PropsWithDefaults): SanitizedProps {
         const locale_format = this.applyDefaultToLocale(props.locale_format);
+        const columns = this.applyDefaultsToColumns(locale_format, props.sort_as_null, props.columns, props.editable);
+        const visibleColumns = this.getVisibleColumns(columns, props.hidden_columns);
+
         let headerFormat = props.export_headers;
         if (props.export_format === 'xlsx' &&  R.isNil(headerFormat)) {
             headerFormat = 'names';
         } else if (props.export_format === 'csv' && R.isNil(headerFormat)) {
             headerFormat = 'ids';
         }
+
         return R.merge(props, {
-            columns: this.applyDefaultsToColumns(locale_format, props.sort_as_null, props.columns, props.editable),
+            columns,
+            export_headers: headerFormat,
             fixed_columns: getFixedColumns(props.fixed_columns, props.row_deletable, props.row_selectable),
             fixed_rows: getFixedRows(props.fixed_rows, props.columns, props.filter_action),
             locale_format,
-            export_headers: headerFormat
+            visibleColumns
```

Comment:
New `visibleColumns` for the TS land -- hidden columns are removed. This is the prop used for most of everything in the table now. `columns` is still used but for specific scenarios.